searchconsole/auth.py

```diff
@@ -64,7 +64,13 @@ def authenticate(client_config, credentials=None, serialize=None):
 
             raise ValueError("Client secrets must be a mapping or path to file")
 
-        flow.run_local_server()
+        if not from_colab:
```

Comment:
Here it would be something like:

```python
auth_flow = InstalledAppFlow.from_client_config(
    ...
)

...

if flow == 'web':
    auth_flow.run_local_server()
elif flow == 'console':
    auth_flow.run_console()
else:
   raise ValueError("Authentication flow '{}' not supported".format(flow))
```